### PR TITLE
chore: Rename histogram buckets to bins

### LINKED ISSUE #281: Update histogram-related "bucket" references to "bin"
To avoid overuse of and possible confusion caused by the term, "bucket," Flux's histogram-related functions and parameters should instead use the term, "bins."

The following functions should be updated:

#### histogram()
The `histogram()` function's `buckets` parameter should be changed to `bins`.
```diff
- histogram(buckets: [0,10,20,+Inf])
+ histogram(bins: [0,10,20,+Inf])
```

#### linearBuckets()
The `linearBuckets()` function should be changed to `linearBins()`.
```diff
- linearBuckets()
+ linearBins()
```

#### logarithmicBuckets()
The `logarithmicBuckets()` function should be changed to `logarithmicBins()`.
```diff
- logarithmicBuckets()
+ logarithmicBins()
```

### PR DESCRIPTION
**Pull request by nathanielc**
Fixes #281

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -1724,9 +1724,9 @@ Histogram has the following properties:
 * `countColumn` string
     CountColumn is the name of the column in which to store the histogram counts.
     Defaults to `_value`.
-* `buckets` array of floats
-    Buckets is a list of upper bounds to use when computing the histogram frequencies.
-    Buckets should contain a bucket whose bound is the maximum value of the data set, this value can be set to positive infinity if no maximum is known.
+* `bins` array of floats
+    Bins is a list of upper bounds to use when computing the histogram frequencies.
+    Bins should contain a bin whose bound is the maximum value of the data set, this value can be set to positive infinity if no maximum is known.
```
Comment (wolffcm) [inline docs/SPEC.md]: Tangential: this sentence is a little confusing.  Maybe `Each element in the array should contain a float value that represents the maximum value for a bin.`

File: docs/SPEC.md [hunk 2]
```diff
@@ -1780,36 +1780,36 @@ Example:
...
 * `count` int
-    Count is the number of buckets to create.
+    Count is the number of bins to create.
 * `inifinity` bool
-    Infinity when true adds an additional bucket with a value of positive infinity.
+    Infinity when true adds an additional bin with a value of positive infinity.
     Defaults to `true`.
 
-#### LogrithmicBuckets
+#### LogrithmicBins
```
Comment (wolffcm) [inline docs/SPEC.md]: Typo: I think the correct spelling is `logarithmic`, unless this is an alternative spelling that I don't know about.

File: functions/transformations/histogram.go [hunk 3]
```diff
@@ -30,15 +30,15 @@ func init() {
...
 		},
-		[]string{"buckets"},
+		[]string{"bins"},
 	)
 
 	flux.RegisterFunction(HistogramKind, createHistogramOpSpec, histogramSignature)
-	flux.RegisterBuiltInValue("linearBuckets", linearBuckets{})
-	flux.RegisterBuiltInValue("logarithmicBuckets", logarithmicBuckets{})
+	flux.RegisterBuiltInValue("linearBins", linearBins{})
+	flux.RegisterBuiltInValue("logarithmicBins", logarithmicBins{})
```
Comment (wolffcm) [inline functions/transformations/histogram.go]: Ok, this looks like the right spelling to me.

**Comment by nathanielc** [comment]
Thanks! Fixed the other typos.

**Pull request merged**